unit-tests-jest/numbers.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Returns a new array that contains only the even numbers in the given array.
+ */
+export function evenNumbers(numbers: number[]): number[] {
+  return numbers.filter((n) => n % 2 === 0);
+}
+
+/**
+ * Returns a number formatted in dollars and cents.
+ */
+export function toDollars(amount: number): string {
+  return `$${amount.toFixed(2)}`;
+}
+
+/**
+ * Returns a new array of numbers where every entry has been divided by the
+ * given divisor. Does not modify the original array.
+ */
+export function divideBy(numbers: number[], divisor: number): number[] {
+  for (let i = 0; i < numbers.length; i++) {
+    numbers[i] = numbers[i] / divisor;
+  }
+  return numbers;
+}
+
+/**
+ * Modifies an object by multiplying the value of each key, but only if that
+ * value is a number. Key names are arbitrary.
+ */
+// we get object literal from main, and we return an object literal with key:string and value:unknown type
+export function multiplyBy(
+  obj: Record<string, unknown>,
+  multiplier: number
+): Record<string, unknown> {
+  const result: Record<string, unknown> = {};
+
+  Object.entries(obj).forEach(([key, value]) => {
+    if (typeof value === 'number') {
+      result[key] = Number(value) * multiplier;
+    }
+    // else if its not a type number return normal key/value; no multiplication
+    else {
+      result[key] = value;
+    }
+  });
+  return result;
```

Comment:
This function is supposed to modify the `obj` object, but it doesn't. It creates a new one.